### DISCUSSION TITLE
On demand mode

### DISCUSSION THREAD

**Discussion by LDprg** [winapps-org/winapps#17] [Ideas]
I had the problem that I don't use winapps every time I start my computer. This produces massive overhead, because I am running a whole windows in the background. For this usecase we should consider the option to starting the vm on demand when a application is called.

**Comment by oskardotglobal** [comment]
Maybe also a timeout after which the VM shuts down after the last application has been closed

**Comment by oskardotglobal** [comment]
Also see #32

**Comment by LDprg** [comment]
With the docker based implementation this should be really easy. Since we can just pause the container and start it later.

We should test if this works as expected though.

First we need to implement the automatic vm creation in the rewrite for docker. Shouldn't be too hard since we can rely on docker compose with the compose file stored in the config folder.

**Comment by coffandro** [comment]
Maybe somehow add an option to enable this when it is? some people will want faster boot times rather than low system usage while others (like myself) don't really care because they need battery life.

**Comment by oskardotglobal** [comment]
This is already a feature that exists
